src/application/preview/Preview.tsx

```diff
@@ -0,0 +1,203 @@
+};
+
+const Preview = ({ onNext, application }: Props): JSX.Element | null => {
+  const [ready, setReady] = useState(false);
+
+  const [ageGroupOptions, setAgeGroupOptions] = useState<{
+    [key: number]: Parameter;
+  }>({});
+  const [purposeOptions, setPurposeOptions] = useState<{
+    [key: number]: Parameter;
+  }>({});
+  const [abilityGroupOptions, setAbilityGroupOptions] = useState<{
+    [key: number]: Parameter;
+  }>({});
+
+  const [reservationUnits, setReservationUnits] = useState<{
+    [key: number]: ReservationUnit;
+  }>({});
+
+  useEffect(() => {
+    async function fetchData() {
+      // there's no api to get reservation units with multiple ids so we're getting them all :)  a.k.a. FIXME
+      const units = await getReservationUnits({ search: undefined });
+      setReservationUnits(
+        mapArrayById(units) as { [key: number]: ReservationUnit }
+      );
+
+      const fetchedAbilityGroupOptions = await getParameters('ability_group');
+      setAbilityGroupOptions(mapArrayById(fetchedAbilityGroupOptions));
+      const fetchedAgeGroupOptions = await getParameters('age_group');
+      setAgeGroupOptions(mapArrayById(fetchedAgeGroupOptions));
+      const fetchedPurposeOptions = await getParameters('purpose');
+      setPurposeOptions(mapArrayById(fetchedPurposeOptions));
+      setReady(true);
+    }
+    fetchData();
+  }, []);
+
+  const { t } = useTranslation();
+
+  const onSubmit = () => {
+    // eslint-disable-next-line no-param-reassign
+    application.status = 'review';
+    onNext();
+  };
+
+  return ready ? (
+    <>
+      <div className={styles.subHeadLine}>
+        {t('Application.preview.basicInfoSubHeading')}
+      </div>
+      <div className={styles.twoColumnContainer}>
+        <LabelValue
+          label={t('Application.preview.firstName')}
+          value={application.contactPerson?.firstName}
+        />
+        <LabelValue
+          label={t('Application.preview.lastName')}
+          value={application.contactPerson?.lastName}
+        />
+        <LabelValue
+          label={t('Application.preview.email')}
+          value={application.contactPerson?.email}
+        />
+      </div>
+      {application.applicationEvents.map((applicationEvent) => (
+        <div key={applicationEvent.id}>
+          <div className={styles.subHeadLine}>{applicationEvent.name}</div>
+          <div className={styles.twoColumnContainer}>
+            <LabelValue
+              label={t('Application.preview.applicationEvent.name')}
+              value={applicationEvent.name}
+            />
+            <LabelValue
+              label={t('Application.preview.applicationEvent.numPersons')}
+              value={applicationEvent.numPersons}
+            />
+            <LabelValue
+              label={t('Application.preview.applicationEvent.ageGroup')}
+              value={`${
+                ageGroupOptions[applicationEvent.ageGroupId || 0].minimum
```

Comment:
Seems 0:th item is not most likey found as the array keys are derived from actual id's.